Mixpanel/PushNotifications.swift

```diff
@@ -0,0 +1,149 @@
+import UIKit
+import UserNotifications
+
+enum PushTapActionType: String {
+    case browser = "browser"
+    case deeplink = "deeplink"
+    case homescreen = "homescreen"
+}
+
+@available(iOS 10.0, *)
+public class MixpanelPushNotifications {
+
+    public static func isMixpanelPushNotification(_ content: UNNotificationContent) -> Bool {
+        return content.userInfo["mp"] != nil
+    }
+
+    @available(iOSApplicationExtension, unavailable)
+    public static func handleResponse(response: UNNotificationResponse,
+           withCompletionHandler completionHandler:
+             @escaping () -> Void) {
+
+        guard self.isMixpanelPushNotification(response.notification.request.content) else {
+            Logger.debug(message: "Calling MixpanelPushNotifications.handleResponse on a non-Mixpanel push notification is a noop...")
+            completionHandler()
+            return
+        }
+        
+        let userInfo = response.notification.request.content.userInfo
+        
+        // Initialize properties to track to Mixpanel
+        var trackingProps: Properties = [:]
+        let mpMetaData = userInfo["mp"] as? [AnyHashable: Any]
+        if mpMetaData != nil {
+            trackingProps["campaign_id"] = mpMetaData!["c"] as! Int
+            trackingProps["message_id"] = mpMetaData!["m"] as! Int
+        }
+
+
+        Logger.debug(message: "%@ didReceiveNotificationResponse action: \(response.actionIdentifier)");
+
+        // If the notification was dismissed, just track and return
+        if response.actionIdentifier == UNNotificationDismissActionIdentifier {
+            for instance in Mixpanel.allInstances() {
+                instance.track(event:"$push_notification_dismissed", properties:trackingProps)
+                instance.flush()
+            }
+            completionHandler();
+            return;
+        }
+
+        var ontap: [AnyHashable: Any]? = nil
+
+        if response.actionIdentifier == UNNotificationDefaultActionIdentifier {
+            // The action that indicates the user opened the app from the notification interface.
+            trackingProps["tap_target"] = "notification";
+
+            if (userInfo["mp_ontap"] != nil) {
```

Comment:
nit: if let ..., 
so you don't have to force wrap.

Reply:
we don't actually force unwrap in this instance